integrations: Add an instruction related to Slack Bridge setup.

PieterCK opened the pull request:
To set up a Slack Bridge, we use generic bots instead of incoming webhook bots for the Slack webhook integration. This PR adds an optional instruction on how to do that if the user is setting up a Slack Bridge.

Part of https://github.com/zulip/python-zulip-api/pull/826

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**
Cross-linking to the Slack bridge:
![image](https://github.com/user-attachments/assets/4bcab0f3-1e3d-4a38-83ef-04e3999944f3)

The new instruction related to Slack Bridge setup is in step 4:
![image](https://github.com/user-attachments/assets/d4993b3e-472b-4369-9b96-d01b438645d9)

The new doc as a whole:
![image](https://github.com/user-attachments/assets/c9c0a25a-16eb-433c-846d-e552d42157e1)

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

timabbott commented on zerver/webhooks/slack/doc.md:
```diff
@@ -50,6 +50,12 @@ See also the [Zulip Slack incoming webhook integration][1].
+   `?api_key=` query in the **integration URL** you generated with the
+   API key of the **Generic bot** used for the Slack Bridge. Make sure
+   to continue the setup with this new URL. Otherwise, proceed to the
+   next step.
```
I think it'd be better to have this at the start, before we ask you to make an incoming webhook bot. Seems annoying to get this far and then need to go make another bot. Something like:

> If you'd like to run a bidirectional bridge, you'll want to create a generic bot ...

PieterCK replied:
ah Ok, my thinking was the user has started the setup by reading the Slack bridge instruction in the python-zulip-api repo first which covers making a generic bot for the Slack bridge before setting up the Slack webhook integration.

PieterCK replied:
I move the instruction to just below step 4

PieterCK replied:
hmm on second thought I think that's not a problem because creating a generic bot _is_ done before the incoming webhook bot. It's step 1 in the Slack bridge setup whereas setting up the Slack webhook integration is step 3.

So in that case I think it makes more sense to have them near the steps where the user is fiddling with the integration URL, like after the instructions around `&channels_map_tp_topics` setting.

I'll mention to use the same generic bot for the Slack bridge though, to help clarify.

alya commented:
I think we need to explain what's going on in order to add these steps. What's a Slack Bridge? Why would I want one or not want one? What happens if I make one? This all needs to be explained here, without requiring the reader to go to https://github.com/zulip/python-zulip-api/blob/main/zulip/integrations/bridge_with_slack/README.md.

alya commented:
But I'm also not following why these instructions are being added here. If you follow these steps, you won't get a Slack bridge, right? So what's the purpose of putting them here?

PieterCK commented:
Sorry looks like I had outdated commits in the PR, so it looks wonky. There were two of the same instructions... :sweat:

PieterCK commented:
Alya [said](https://github.com/zulip/zulip/pull/32714#issuecomment-2613105974):
>I think we need to explain what's going on in order to add these steps. What's a Slack Bridge? Why would I want one or not want one? What happens if I make one? This all needs to be explained here, without requiring the reader to go to https://github.com/zulip/python-zulip-api/blob/main/zulip/integrations/bridge_with_slack/README.md.

hmm, I think the users who are looking to set up Slack bridge would start at the [Slack bridge page](https://github.com/zulip/python-zulip-api/blob/main/zulip/integrations/bridge_with_slack/README.md) instead of at the Slack webhook integration page. And if they do, it's at step 3 in the Slack bridge documentation where we instruct the user to set up a Slack webhook integration, so by that time I would think that they already have the context. 

I added this to help clarify things. (the second paragraph):
![image](https://github.com/user-attachments/assets/88263369-4a15-45eb-b751-09e5e3ecc9e3)

PieterCK commented:
Alya [said](https://github.com/zulip/zulip/pull/32714#issuecomment-2613107878):
>But I'm also not following why these instructions are being added here. If you follow these steps, you won't get a Slack bridge, right? So what's the purpose of putting them here?

Because setting up the Slack webhook integration is part of setting up the Slack bridge and they need to set up the Slack webhook integration in a slightly different way (use **Generic bot** instead) if it's for the Slack bridge, so that's what the optional step is for.

PieterCK commented:
Followed up on a comment from Tim [here](https://github.com/zulip/zulip/pull/32714#discussion_r1930313419)

PieterCK commented:
Thanks for taking a look at this! I've addressed your comments, pushed new changes and updated the screenshot in the description, this is ready for you to take a look again! @alya

timabbott commented on zerver/webhooks/slack/doc.md:
```diff
@@ -31,6 +34,11 @@ See also the [Zulip Slack incoming webhook integration][1].
+1. If you're setting up a **Slack Bridge**, switch the value of the
+   `?api_key=` query in the **integration URL** you generated with the
+   API key of the **Generic bot** used for the Slack Bridge. Make sure
+   to continue the setup with this new URL. Otherwise, skip this step.
```
"Switch" is confusing language, since it usually implies a swap, i.e. putting the value you're replacing where you got it from. Do you mean "change the value"

alya commented:
Thanks, this was a helpful start! Closing in favor of a more comprehensive edit pass in #33211.
